src/store/singing.ts

```diff
@@ -2936,12 +2945,236 @@ export const singingStore = createPartialStore<SingingStoreTypes>({
+              .filter((value) => value.trackId === trackId)
+              .filter((value) => value.queryKey != undefined)
+              .toSorted((a, b) => a.startTime - b.startTime);
+
+            if (phrases.length === 0) {
+              continue;
+            }
+
+            const phraseQueries = phrases.map((value) => {
+              const phraseQuery =
+                value.queryKey != undefined
+                  ? state.phraseQueries.get(value.queryKey)
+                  : undefined;
+              if (phraseQuery == undefined) {
+                throw new Error("phraseQuery is undefined.");
+              }
+              return phraseQuery;
+            });
+            const phraseStartTimes = phrases.map((value) => value.startTime);
+
+            for (const phraseQuery of phraseQueries) {
+              // フレーズのクエリのフレームレートとエディターのフレームレートが一致しない場合はエラー
+              // TODO: 補間するようにする
+              if (phraseQuery.frameRate != frameRate) {
+                throw new Error(
+                  "The frame rate between the phrase query and the editor does not match.",
+                );
+              }
+            }
+
+            const phraseStartFrames = calcPhraseStartFrames(
+              phraseStartTimes,
+              frameRate,
+            );
+            const phraseEndFrames = calcPhraseEndFrames(
+              phraseStartFrames,
+              phraseQueries,
+            );
+
+            const phrasePhonemeSequences = phraseQueries.map((query) => {
+              return query.phonemes;
+            });
+            const entirePhonemeTimings = toEntirePhonemeTimings(
+              phrasePhonemeSequences,
+              phraseStartFrames,
+            );
+
+            // TODO: 音素タイミング編集データを取得して適用するようにする
+
+            adjustPhonemeTimingsAndPhraseEndFrames(
+              entirePhonemeTimings,
+              phraseStartFrames,
+              phraseEndFrames,
+            );
+
+            // 一番最初のpauseの開始フレームの値が0より大きい場合は0にする
+            if (entirePhonemeTimings.length === 0) {
+              throw new Error("entirePhonemeTimings.length is 0.");
+            }
+            if (entirePhonemeTimings[0].startFrame > 0) {
+              entirePhonemeTimings[0].startFrame = 0;
+            }
+
+            // マイナス時間のものを除く（マイナス時間のところは書き出さない）
+            for (const phonemeTiming of entirePhonemeTimings) {
+              if (phonemeTiming.startFrame < 0) {
+                phonemeTiming.startFrame = 0;
+              }
+              if (phonemeTiming.endFrame < 0) {
+                phonemeTiming.endFrame = 0;
+              }
+            }
+            const filteredEntirePhonemeTimings = entirePhonemeTimings.filter(
+              (value) => {
+                const frameLength = value.endFrame - value.startFrame;
+                return frameLength >= 1;
+              },
+            );
+
+            const entirePhonemes = phonemeTimingsToPhonemes(
+              filteredEntirePhonemeTimings,
+            );
```

Comment:
改行位置にちょっと違和感があるかも＆コメントが有っても良いかも
こんな感じ？
```suggestion

            // フレーム数が1未満の音素を除く
            const filteredEntirePhonemeTimings = entirePhonemeTimings.filter(
              (value) => {
                const frameLength = value.endFrame - value.startFrame;
                return frameLength >= 1;
              },
            );

            const entirePhonemes = phonemeTimingsToPhonemes(
              filteredEntirePhonemeTimings,
            );
```

Reply:
改行位置とコメントを修正しました！